testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -103,15 +111,39 @@ impl Experiment for PerformanceBenchmark {
         instance::instancelist_to_set(&self.down_instances)
     }
 
-    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+    async fn run(&mut self, context: &mut Context<'_>) -> Result<()> {
         let stop_effects: Vec<_> = self
             .down_instances
             .clone()
             .into_iter()
             .map(StopContainer::new)
             .collect();
-        let futures = stop_effects.iter().map(|e| e.activate());
-        join_all(futures).await;
+        if let Some(cluster_swarm) = context.cluster_swarm {
+            let instance_configs: Vec<&InstanceConfig> = self
+                .down_instances
+                .iter()
+                .map(|instance| -> Result<&InstanceConfig> {
+                    instance
+                        .instance_config()
+                        .ok_or_else(|| format_err!("Failed to find instance_config"))
+                })
+                .collect::<Result<_, _>>()?;
+            let futures: Vec<_> = instance_configs
+                .into_iter()
+                .map(|ic| match ic {
+                    Validator(validator_config) => {
+                        cluster_swarm.delete_validator(validator_config.clone())
```

Comment:
Should we just make `cluster_swarm.delete_node(Config)` function?

Reply:
Done